cmd/cli/uninstall_mesh.go

```diff
@@ -199,6 +182,63 @@ func (d *uninstallMeshCmd) run() error {
 			fmt.Fprintf(d.out, "OSM namespace CANNOT be deleted in a managed environment\n")
 		}
 	}
+
+	err := d.deleteClusterResources()
+	return err
+}
+
+func (d *uninstallMeshCmd) findFlagSpecifiedMesh(meshInfoList []meshInfo) (bool, bool) {
+	meshFlagSpecified := d.meshName != ""
+	specifiedMeshFound := false
+	if !meshFlagSpecified {
+		return meshFlagSpecified, specifiedMeshFound
+	}
+
+	specifiedMeshFound = d.findMesh(meshInfoList)
+	if !specifiedMeshFound {
+		fmt.Fprintf(d.out, "Did not find mesh [%s] in namespace [%s]\n", d.meshName, d.meshNamespace)
+		// print a list of meshes within the cluster for a better user experience
+		if err := d.printMeshes(); err != nil {
+			fmt.Fprintf(d.out, "Unable to list meshes in the cluster - [%v]", err)
+		}
+	}
+
+	return meshFlagSpecified, specifiedMeshFound
+}
+
+func (d *uninstallMeshCmd) promptMeshUninstall(meshInfoList, meshesToUninstall []meshInfo, meshFlagSpecified bool) ([]meshInfo, error) {
+	for _, mesh := range meshInfoList {
+		// Only prompt for specified mesh if `mesh-name` is specified
+		if meshFlagSpecified && mesh.name != d.meshName {
+			continue
+		}
+		confirm, err := confirm(d.in, d.out, fmt.Sprintf("\nUninstall OSM [mesh name: %s] in namespace [%s] and/or OSM resources?", mesh.name, mesh.namespace), 3)
+		if err != nil {
+			return nil, err
+		}
+		if confirm {
+			meshesToUninstall = append(meshesToUninstall, mesh)
+		}
+	}
+	return meshesToUninstall, nil
+}
+
+func (d *uninstallMeshCmd) deleteNs(ctx context.Context, ns string) error {
+	if !d.deleteNamespace {
+		return nil
+	}
+	if err := d.clientSet.CoreV1().Namespaces().Delete(ctx, ns, v1.DeleteOptions{}); err != nil {
+		if k8sApiErrors.IsNotFound(err) {
+			fmt.Fprintf(d.out, "OSM namespace [%s] not found\n", ns)
+			return nil
+		}
+		return errors.Errorf("Error occurred while deleting OSM namespace [%s] - %v", ns, err)
```

Comment:
Its better to lowercase the error message here.

Reply:
I ended up changing the error message to `return errors.Errorf("Could not delete OSM namespace [%s] - %v", ns, err)`